### PR TITLE
WIP: extract the kinto client functionalities in its own module

### PR DESCRIPTION
**Pull request by magopian**
Replaces #11 (with a local branch to ease the development with @n1k0 instead of pushing to my own repository/branch)

Please refer to #11 for some previous comments/reviews

### CODE REVIEW

File: src/Kinto.elm [hunk 1]
```diff
@@ -151,8 +151,14 @@ performQuery config endpoint verb =
...
     in
-        (Http.send Http.defaultSettings request)
-            |> toKintoResponse
+        ((Http.send Http.defaultSettings request)
+            |> Task.mapError NetworkError
+        )
+            `Task.andThen`
+                (\response ->
+                    toKintoResponse response
+                        |> Task.fromResult
```
Comment (glasserc) [inline src/Kinto.elm]: I think instead of using a lambda here, you can do `toKintoResponse >> Task.fromResult`.

File: src/Kinto.elm [hunk 2]
```diff
@@ -211,18 +217,29 @@ errorDecoder =
...
-        |> Task.mapError NetworkError
-    )
-        `Task.andThen` handleResponse
+toKintoResponse : Http.Response -> Result Error Json.Value
+toKintoResponse ({ value, status, statusText } as response) =
+    (extractBody response)
+        `Result.andThen`
+            (\body ->
+                extractData body
+                    |> Result.formatError (extractError status statusText)
```
Comment (glasserc) [inline src/Kinto.elm]: Same here, I think you can use functional composition instead of a lambda.

Comment (magopian) [reply src/Kinto.elm]: Awesome! Thanks a lot, didn't know there was a function composition in elm (thought I should have guessed, as a functional language :P

**Comment by magopian** [comment]
Thanks for the tip on function composition @glasserc, so does that current version look more in line with what you had in mind when commenting https://github.com/n1k0/kinto-elm-experiments/pull/11#discussion_r82228870 ?

**Comment by magopian** [comment]
Well, seems it's good to land now, thanks for all the feedback and help @glasserc !

**Pull request merged**